File: Cpp/492_Construct_the_Rectangle.cpp
class Solution {
public:
    vector<int> constructRectangle(int area) {
        vector<int> result;
        int square_root = std::sqrt(area);
        for(int i = square_root; i>0; --i){
            if(area % i == 0) {
                result.push_back(area/i);
                result.push_back(i);
                break;
            }
        }
        return result;   
    }
};

File: Kotlin/492_Construct_the_Rectangle.kt
class Solution {
    fun constructRectangle(area: Int): IntArray {
        val squareRoot = Math.sqrt(area.toDouble()).toInt()
        for(i in squareRoot downTo 1){
            if(area % i == 0) {
                return intArrayOf(area/i, i)
            }
        }
        
        return intArrayOf()
    }
}